vividus-plugin-saucelabs/src/main/java/org/vividus/saucelabs/SauceLabsTestStatusManager.java

```diff
@@ -24,18 +31,37 @@
 
 public class SauceLabsTestStatusManager extends AbstractCloudTestStatusManager
 {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SauceLabsTestStatusManager.class);
+
     private final JavascriptActions javascriptActions;
+    private final SauceREST sauceREST;
 
     public SauceLabsTestStatusManager(IWebDriverProvider webDriverProvider, TestContext testContext,
-            JavascriptActions javascriptActions)
+            JavascriptActions javascriptActions, SauceREST sauceREST)
     {
         super(new CloudTestStatusMapping("passed", "failed"), webDriverProvider, testContext);
         this.javascriptActions = javascriptActions;
+        this.sauceREST = sauceREST;
     }
 
     @Override
     protected void updateCloudTestStatus(String sessionId, String status) throws UpdateCloudTestStatusException
     {
         javascriptActions.executeScript("sauce:job-result=" + status);
     }
+
+    @Override
+    protected boolean isTestExecutionComplete(String sessionId)
```

Comment:
should I throw specific exception here? e.g. UnableToGetCloudExecutionStatusException?